### PR TITLE
Support position for devtool

### PR DESCRIPTION
**Pull request by jellydn**
## WHAT
- Migrate example from create-react-app to vitejs
- Support 4 positions for devtool UI: top-left, top-right, bottom-left, bottom-right. And keep `top-right` as default option
- Add storybook for easy to test/prototype component
- Update packages

Storybook:
https://react-hook-form-devtool-ui.vercel.app/?path=/story/devtoolui--default-dev-tool

## WHY
- We should support `position` as a prop because it's easy to adjust for different UI.

## HOW

- New usage

```
<DevTool control={control} placement="top-left" />
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] Linter
- [x] Tests
- [ ] Review comments
- [ ] Security

### CODE REVIEW

**Comment by bluebill1049** [comment]
oh wow, thanks very much @jellydn  this is awesome, I will contribute to the UI as well after this is merged. I will bake those buttons inside the pannel so they look nicer.

**Comment by bluebill1049** [comment]
Let me know it's ready to be reviewed.

**Comment by jellydn** [comment]
Thanks Bill. It's ready. Please take a look. Thanks 😀

File: example/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (bluebill1049) [inline example/package.json:1]: nice! thanks for the huge improvement!

**Review by bluebill1049 — APPROVED** [review]
LTGM, let's get it in and i will polish up the UI part.

File: example/package.json (hunk 1 above)
Comment (jellydn) [reply example/package.json:1]: Great. Thanks. Good to hear that.

**Comment by bluebill1049** [comment]
sorry about the delay, and thanks for your hard work. I am publishing it now.

**Comment by bluebill1049** [comment]
let me know your twitter acc, i can tag you for the release.

**Comment by jellydn** [comment]
Thanks @bluebill1049 This is my twitter @jellydn

**Comment by bluebill1049** [comment]
https://twitter.com/HookForm/status/1391665467329105920 really appreciated your work on the storybook as well!

**Comment by bluebill1049** [comment]
if i find some more free time, I will do a "setting" icon in the panel as well. so user can switch it around.

**Comment by jellydn** [comment]
That would be great. Thanks.

**Pull request merged**